lib/OpenQA/Schema/ResultSet/Jobs.pm

```diff
@@ -215,203 +210,135 @@ sub create_from_settings {
-        if ($key eq 'WORKER_CLASS') {
-            @values = split(m/,/, $concatenated_value);
-        }
-        else {
-            @values = ($concatenated_value);
-        }
-
-        for my $value (@values) {
-            push(@job_settings, {t_created => $now, t_updated => $now, key => $key, value => $value});
-        }
+        my @values = $key eq 'WORKER_CLASS' ? split(m/,/, $settings{$key}) : ($settings{$key});
+        push(@job_settings, {t_created => $now, t_updated => $now, key => $key, value => $_}) for (@values);
     }
     $job->settings->populate(\@job_settings);
 
     # associate currently available assets with job
     $job->register_assets_from_settings;
 
-    if (%group_args && !$group) {
-        log_warning('Ignoring invalid group ' . encode_json(\%group_args) . ' when creating new job ' . $job->id);
-    }
-
+    log_warning('Ignoring invalid group ' . encode_json(\%group_args) . ' when creating new job ' . $job->id)
+      if %group_args && !$group;
     $job->calculate_blocked_by;
-
     $txn_guard->commit;
     return $job;
 }
 
-sub complex_query {
-    my ($self, %args) = @_;
-
-    # For args where we accept a list of values, allow passing either an
-    # array ref or a comma-separated list
-    for my $arg (qw(state ids result failed_modules modules modules_result)) {
-        next                                    unless $args{$arg};
-        $args{$arg} = [split(',', $args{$arg})] unless (ref($args{$arg}) eq 'ARRAY');
-    }
-
+sub prepare_complex_query_search_args ($self, $args) {
     my @conds;
-    my %attrs;
     my @joins;
 
-    if (my $columns = $args{columns}) {
-        $attrs{columns} = $columns;
-    }
-    if (my $prefetch = $args{prefetch}) {
-        $attrs{prefetch} = $prefetch;
-    }
-
-    if ($args{failed_modules}) {
-        push @joins, "modules";
-        push(
-            @conds,
-            {
-                'modules.name'   => {-in => $args{failed_modules}},
-                'modules.result' => OpenQA::Jobs::Constants::FAILED,
-            });
-    }
-    if ($args{modules}) {
-        push @joins, "modules";
-        push(
-            @conds,
-            {
-                'modules.name' => {-in => $args{modules}}});
-    }
-    if ($args{modules_result}) {
-        push(@joins, "modules") unless grep { $_ eq "modules" } @joins;
-        push(
-            @conds,
-            {
-                'modules.result' => {-in => $args{modules_result}}});
-    }
-
-    if ($args{state}) {
-        push(@conds, {'me.state' => $args{state}});
-    }
-    if ($args{maxage}) {
-        my $agecond = {'>' => time2str('%Y-%m-%d %H:%M:%S', time - $args{maxage}, 'UTC')};
-        push(
-            @conds,
-            {
-                -or => [
-                    'me.t_created'  => $agecond,
-                    'me.t_started'  => $agecond,
-                    'me.t_finished' => $agecond
-                ]});
+    if ($args->{failed_modules}) {
+        push @joins, 'modules';
```

Comment:
this is interesting to see that this code section is not covered. Means we can completely delete it and no tests will fail. Maybe this is actually obsoleted and replaced by the query for "modules" together with "modules_result". Should we just delete this? Interesting though is that "failed_modules" as a word does appear in tests.

EDIT: I have now added three more commits deleting dead code or at least uncovered code

Reply:
Seems it was removed from the HTML form in 4865792863e7f298a23b3640052c748b05d265e0 but forgotten in the backend

Reply:
ah, perfect! I will include that reference in the git commit message. Thank you.